pilot/pkg/model/authorization.go

```diff
@@ -30,162 +29,168 @@ type RolesAndBindings struct {
 	// ServiceRoles in the same namespace.
 	Roles []Config
 
-	// Maps from ServiceRole name to its associated ServiceRoleBindings.
-	RoleNameToBindings map[string][]*rbacproto.ServiceRoleBinding
+	// ServiceRoleBindings indexed by its associated ServiceRole's name.
+	Bindings map[string][]*rbacproto.ServiceRoleBinding
 }
 
-// AuthorizationPolicyConfig stores the AuthorizationPolicy and its name.
-type AuthorizationPolicyConfig struct {
-	Name string
+// AuthorizationPolicies organizes authorization policies by namespace.
+type AuthorizationPolicies struct {
+	// Maps from namespace to RolesAndBindings.
+	namespaceToV1Policies map[string]*RolesAndBindings
+
+	// The mesh global RbacConfig.
+	rbacConfig *rbacproto.RbacConfig
 }
 
-// AuthorizationConfigV2 stores a list of AuthorizationPolicyConfig and ServiceRole in a given namespace.
-type AuthorizationConfigV2 struct {
-	// A list of AuthorizationPolicyConfig.
-	AuthzPolicies []*AuthorizationPolicyConfig
+// NewAuthorizationPolicies creates a new AuthorizationPolicies.
+func NewAuthorizationPolicies(env *Environment) (*AuthorizationPolicies, error) {
+	policy := &AuthorizationPolicies{
+		namespaceToV1Policies: map[string]*RolesAndBindings{},
+	}
+
+	rbacConfig := env.IstioConfigStore.ClusterRbacConfig()
+	if rbacConfig == nil {
+		rbacConfig = env.IstioConfigStore.RbacConfig()
+	}
+	if rbacConfig != nil {
```

Comment:
If rbacConfig is nil, do we need to extract policy?

Reply:
we don't really need to if rbacConfig is nil, we already cover this situation when generating envoy config later.